### PR TITLE
Home page analytics UI with required fixes to analytics for litmus-portal.

### PR DESCRIPTION
**Pull request by ishangupta-ds**
## Proposed changes

Adds home page analytics components to be integrated.

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- PR #2222 .

## Special notes for your reviewer:
Analytics PR#2 (post-RC)

### CODE REVIEW

File: litmus-portal/frontend/src/views/Home/ReturningHome/style.ts [hunk 1]
```diff
@@ -14,6 +12,88 @@ const useStyles = makeStyles((theme) => ({
...
+
+  extrasDiv: {
+    marginRight: theme.spacing(2.5),
+  },
+
+  fixedRecents: {
+    height: '15rem',
+    width: '20rem',
+    color: theme.palette.text.secondary,
+    // backgroundColor: theme.palette.secondary.contrastText,
```
Comment (A4k9) [inline litmus-portal/frontend/src/views/Home/ReturningHome/style.ts:83]: Is this comment going to be used later? If not, please consider removing.

File: litmus-portal/frontend/src/views/Home/ReturningHome/style.ts [hunk 2]
```diff
@@ -14,6 +12,88 @@ const useStyles = makeStyles((theme) => ({
...
+    background: theme.palette.customColors.black(0.7),
+  },
+
+  backgroundFix: {
+    color: theme.palette.text.secondary,
+    backgroundColor: theme.palette.secondary.contrastText,
+  },
+
+  commingSoon: {
+    color: '#FFF',
```
Comment (A4k9) [inline litmus-portal/frontend/src/views/Home/ReturningHome/style.ts:93]: Can a similar colour from the theme file be used here? Please look into it.

**Comment by ishangupta-ds** [comment]
This PR is not ready for review @A4k9 PTAL at the first PR for analytics https://github.com/litmuschaos/litmus/pull/2222 PR #2222

File: litmus-portal/frontend/src/views/Home/ReturningHome/style.ts (hunk 2 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/Home/ReturningHome/style.ts:93]: This is a draft PR @A4k9 many changes will go in, waiting for #2222 to get merged as this one has a data dependency on it.

File: litmus-portal/frontend/src/views/Home/ReturningHome/style.ts (hunk 1 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/Home/ReturningHome/style.ts:83]: It won't be there at all in final commit to this PR you can take a look at PR #2222  @A4k9

Comment (A4k9) [reply litmus-portal/frontend/src/views/Home/ReturningHome/style.ts:83]: Ok thanks!

File: litmus-portal/frontend/src/views/Home/RecentActivity/RecentActivityListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+const RecentActivityListItem: React.FC<RecentActivityListItemProps> = ({
+  message,
+}) => {
+  const classes = useStyles();
+  return (
+    <ListItem>
+      <ListItemAvatar>
+        <Typography className={classes.messageID}>
+          {' '}
+          &nbsp; &nbsp; {message.id}
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/views/Home/RecentActivity/RecentActivityListItem.tsx:25]: please do not hardcode spacing into html

File: litmus-portal/frontend/src/views/Home/RecentActivity/style.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    height: '14rem',
+    '&::-webkit-scrollbar': {
+      width: '0.2em',
+    },
+    '&::-webkit-scrollbar-track': {
+      webkitBoxShadow: `inset 0 0 6px ${theme.palette.common.black}`,
+    },
+    '&::-webkit-scrollbar-thumb': {
+      backgroundColor: theme.palette.secondary.dark,
+      outline: '1px solid slategrey',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/views/Home/RecentActivity/style.ts:16]: take from theme

File: litmus-portal/frontend/src/views/Home/ResilienceScoreComparisonPlot/index.tsx [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+            value={currentGranularity.name}
+            onChange={handleChangeInGranularity}
+            label="Granularity"
+            inputProps={{
+              name: 'name',
+              id: 'outlined-selection-granularity',
+            }}
+            className={classes.root}
+          >
+            <MenuItem value="Hourly">Hourly</MenuItem>
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/views/Home/ResilienceScoreComparisonPlot/index.tsx:307]: take these from translation

**Comment by ishangupta-ds** [comment]
@arkajyotiMukherjee @S-ayanide @A4k9 @Saranya-jena @oumkale @Jonsy13 @amityt PTAL. cc: @rajdas98 @ajeshbaby

File: litmus-portal/frontend/src/views/Home/ResilienceScoreComparisonPlot/index.tsx [hunk 6]
```diff
@@ -0,0 +1,357 @@
...
+            autosizable: true,
+            responsive: true,
+            frameMargins: 0.2,
+            showAxisDragHandles: true,
+            showAxisRangeEntryBoxes: true,
+            showTips: true,
+            displayModeBar: false,
+            toImageButtonOptions: {
+              format: 'png',
+              filename: 'Top4_ResilienceScores_Comparison',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/views/Home/ResilienceScoreComparisonPlot/index.tsx:345]: I hope this the file name to export the plot...if that is so...it needs to be unique each time

File: litmus-portal/frontend/src/views/Home/ReturningHome/style.ts [hunk 7]
```diff
@@ -14,6 +12,88 @@ const useStyles = makeStyles((theme) => ({
...
+
+  backgroundFix: {
+    color: theme.palette.text.secondary,
+    backgroundColor: theme.palette.secondary.contrastText,
+  },
+
+  commingSoon: {
+    color: theme.palette.secondary.contrastText,
+    marginLeft: '17%',
+    marginTop: '30%',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/views/Home/ReturningHome/style.ts:95]: don't give margins in percentage

**Review by arkajyotiMukherjee — DISMISSED** [review]

File: litmus-portal/frontend/src/pages/AnalyticsPage/index.tsx [hunk 8]
```diff
@@ -87,10 +89,13 @@ const AnalyticsPage: React.FC = () => {
...
-  // Query to get workflows
-  const { data, error } = useQuery<Workflow, WorkflowDataVars>(
-    WORKFLOW_DETAILS,
-    { variables: { projectID: selectedProjectID } }
+  // Apollo query to get the scheduled workflow data
+  const { data, error } = useQuery<WorkflowList, WorkflowListDataVars>(
+    WORKFLOW_LIST_DETAILS,
+    {
+      variables: { projectID: selectedProjectID, workflowIDs: [] },
+      fetchPolicy: 'cache-and-network',
```
[the remark this answers is not in the document]
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/pages/AnalyticsPage/index.tsx:97]: This can be removed..

File: litmus-portal/frontend/src/views/Home/RecentActivity/RecentActivityListItem.tsx (hunk 3 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/Home/RecentActivity/RecentActivityListItem.tsx:25]: Changing it..

File: litmus-portal/frontend/src/views/Home/ResilienceScoreComparisonPlot/index.tsx (hunk 6 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/Home/ResilienceScoreComparisonPlot/index.tsx:345]: Okay attaching a timestamp to it.

File: litmus-portal/frontend/src/pages/AnalyticsPage/styles.ts [hunk 9]
```diff
@@ -49,6 +49,11 @@ const useStyles = makeStyles((theme) => ({
   loader: {
     marginLeft: '-8%',
```
Comment (Saranya-jena) [inline litmus-portal/frontend/src/pages/AnalyticsPage/styles.ts:50]: Can you use theme.spacing?

File: litmus-portal/frontend/src/views/Home/RecentActivity/index.tsx [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+const RecentActivity: React.FC<RecentActivityProps> = ({ activities }) => {
+  const classes = useStyles();
+
+  return (
+    <List dense className={classes.tabContainer}>
+      {activities.length === 0 ? (
+        <ListItem>
+          <ListItemText>No recent activity.</ListItemText>
```
Comment (Saranya-jena) [inline litmus-portal/frontend/src/views/Home/RecentActivity/index.tsx:19]: Can you put this in translation?

File: litmus-portal/frontend/src/views/Home/ResilienceScoreComparisonPlot/index.tsx [hunk 11]
```diff
@@ -0,0 +1,357 @@
...
+          variant="outlined"
+          className={classes.formControl}
+          color="secondary"
+          focused
+        >
+          <InputLabel
+            htmlFor="outlined-selection-granularity"
+            className={classes.root}
+          >
+            Granularity
```
Comment (Saranya-jena) [inline litmus-portal/frontend/src/views/Home/ResilienceScoreComparisonPlot/index.tsx:295]: add it to translation

File: litmus-portal/frontend/src/views/Home/ResilienceScoreComparisonPlot/index.tsx [hunk 12]
```diff
@@ -0,0 +1,357 @@
...
+            />
+          </IconButton>
+        </Tooltip>
+      </div>
+      <div className={classes.plot}>
+        <Plot
+          data={plotData}
+          layout={plotLayout}
+          useResizeHandler
+          style={{
```
Comment (Saranya-jena) [inline litmus-portal/frontend/src/views/Home/ResilienceScoreComparisonPlot/index.tsx:330]: Can you put this in styles?

**Review by Saranya-jena — DISMISSED** [review]

File: litmus-portal/frontend/src/views/Home/ResilienceScoreComparisonPlot/index.tsx (hunk 12 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/Home/ResilienceScoreComparisonPlot/index.tsx:330]: I tried this out but it didn't work..

File: litmus-portal/frontend/src/pages/HomePage/index.tsx [hunk 13]
```diff
@@ -182,36 +182,40 @@ const HomePage: React.FC = () => {
...
-                      }}
-                    >
-                      <div className={classes.btnSpan}>
-                        <Typography className={classes.btnText}>
-                          {t('home.analytics.moreInfo')}
-                        </Typography>
-                        <img src="icons/next.png" alt="next" />
-                      </div>
-                    </Button>
+                      <Button
```
Comment (oumkale) [inline litmus-portal/frontend/src/pages/HomePage/index.tsx:193]: Please will use the present Button component?

Comment (ishangupta-ds) [reply litmus-portal/frontend/src/pages/HomePage/index.tsx:193]: It was already there @oumkale

File: litmus-portal/frontend/src/components/Header/index.tsx [hunk 14]
```diff
@@ -175,13 +171,12 @@ const Header: React.FC = () => {
                 <CustomBreadCrumbs location={useLocation().pathname} />
               </Box>
               <Box p={1} className={classes.headerFlexPadded}>
-                {/* uncomment to activate notifications.
-                  <NotificationsDropdown
+                {/*              <NotificationsDropdown
                   count={`${countOfMessages}`}
                   messages={messages}
```
Comment (oumkale) [inline litmus-portal/frontend/src/components/Header/index.tsx:176]: You may remove this commented code

File: litmus-portal/frontend/src/pages/HomePage/index.tsx (hunk 13 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/pages/HomePage/index.tsx:193]: I haven't made changes to existing components used, this can go in as separate bug-fix/enhancement PR.

File: litmus-portal/frontend/src/components/Header/index.tsx (hunk 14 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/Header/index.tsx:176]: This was intentionally left to keep some info about the props, interface and functions to be used when we have the data for notifications. It was also there from beforehand. Can be removed with another PR if the team agrees. @oumkale

**Comment by ishangupta-ds** [comment]
@arkajyotiMukherjee and @Saranya-jena PTAL, all suggested changes have been made.

**Review by oumkale — DISMISSED** [review]

File: litmus-portal/frontend/src/components/Header/index.tsx (hunk 14 above)
Comment (gdsoumya) [reply litmus-portal/frontend/src/components/Header/index.tsx:176]: Remove the notifications comment, will add it later when the feature is ready

**Review by gdsoumya — CHANGES_REQUESTED** [review]

File: litmus-portal/frontend/src/pages/HomePage/index.tsx (hunk 13 above)
Comment (arkajyotiMukherjee) [reply litmus-portal/frontend/src/pages/HomePage/index.tsx:193]: can you fix this in this PR itself @ishangupta-ds ?

File: litmus-portal/frontend/src/components/Header/index.tsx (hunk 14 above)
Comment (arkajyotiMukherjee) [reply litmus-portal/frontend/src/components/Header/index.tsx:176]: this can be removed @ishangupta-ds

File: litmus-portal/frontend/src/pages/HomePage/index.tsx (hunk 13 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/pages/HomePage/index.tsx:193]: The styles are breaking @arkajyotiMukherjee on using that, also I am having trouble with the analytics data in place, that component is not getting displayed half of the times. This we can fix with another PR, much style changes are required there..

File: litmus-portal/frontend/src/components/Header/index.tsx (hunk 14 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/Header/index.tsx:103]: Removed it.

File: litmus-portal/frontend/src/pages/AnalyticsPage/styles.ts [hunk 15]
```diff
@@ -41,13 +41,18 @@ const useStyles = makeStyles((theme) => ({
 
   waitingText: {
     fontSize: '2rem',
-    marginLeft: '25%',
-    marginTop: '25%',
+    marginLeft: '30%',
+    marginTop: '20%',
```
Comment (A4k9) [inline litmus-portal/frontend/src/pages/AnalyticsPage/styles.ts:45]: Please try using theme.spacing() here.

File: litmus-portal/frontend/public/locales/en/translation.yaml [hunk 16]
```diff
@@ -67,6 +67,15 @@ home:
   analytics:
     heading: How busy is the Litmus Project?
     moreInfo: See more
+  commingSoon: Comming soon!
```
Comment (A4k9) [inline litmus-portal/frontend/public/locales/en/translation.yaml:70]: Typo: commingSoon -> comingSoon, same on the actual text

File: litmus-portal/frontend/src/pages/AnalyticsPage/styles.ts (hunk 15 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/pages/AnalyticsPage/styles.ts:45]: I tried this but have to add breakpoints then just for this loading message. If I hardcode the spacing it won't be as responsive as using `%` for the components which are required all have spacing from the theme itself, this is just an error message. @A4k9

**Review by oumkale — APPROVED** [review]

**Review by amityt — APPROVED** [review]

**Review by Jonsy13 — APPROVED** [review]

File: litmus-portal/frontend/src/components/Button/ButtonFilled/styles.ts [hunk 17]
```diff
@@ -22,6 +22,12 @@ const useStyles = makeStyles((theme) => ({
       backgroundColor: theme.palette.primary.dark,
     },
   },
+  buttonTransparent: {
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/Button/ButtonFilled/styles.ts:25]: why were these changes required...can you post screenshots of the button you are trying to make

**Review by arkajyotiMukherjee — CHANGES_REQUESTED** [review]

File: litmus-portal/frontend/src/components/Button/ButtonFilled/styles.ts (hunk 17 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/Button/ButtonFilled/styles.ts:25]: sure.

Comment (arkajyotiMukherjee) [reply litmus-portal/frontend/src/components/Button/ButtonFilled/styles.ts:25]: cc: @S-ayanide

**Comment by ishangupta-ds** [comment]
Screenshot 2020-10-12 at 2 18 25 PM
 see more and see pre-defined workflows @arkajyotiMukherjee

File: litmus-portal/frontend/src/components/Button/ButtonFilled/styles.ts (hunk 17 above)
Comment (arkajyotiMukherjee) [reply litmus-portal/frontend/src/components/Button/ButtonFilled/styles.ts:25]: as per my understanding of the designs...filled button doesn't need to be transparent.

**Comment by arkajyotiMukherjee** [comment]
> Screenshot 2020-10-12 at 2 18 25 PM
> 
> [see more] and [see pre-defined workflows] @arkajyotiMukherjee

alright the see more button is not actually a button...it's just link
so a div with a text and an icon with an onClick event will suffice

**Comment by amityt** [comment]
@arkajyotiMukherjee  I think the "See more" is a button. We recently fixed that with the Hacktober PR.

**Comment by arkajyotiMukherjee** [comment]
> @arkajyotiMukherjee  I think the "See more" is a button. We recently fixed that with the Hacktober PR.

I've asked ishan to change it to an IconButton as it will not have additional styles

**Comment by amityt** [comment]
> > @arkajyotiMukherjee  I think the "See more" is a button. We recently fixed that with the Hacktober PR.
> 
> I've asked ishan to change it to an IconButton as it will not have additional styles

Yes @arkajyotiMukherjee  That can be done!

**Comment by ishangupta-ds** [comment]
changes made @arkajyotiMukherjee @amityt PTAL.

**Review by oumkale — APPROVED** [review]

**Review by arkajyotiMukherjee — APPROVED** [review]

**Review by amityt — APPROVED** [review]

**Review by Jonsy13 — APPROVED** [review]
